File: src/main/java/ru/uvarov/flashcards/FlashcardsApplication.java
package ru.uvarov.flashcards;

import java.io.IOException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
@Slf4j
public class FlashcardsApplication {

    public static void main(String[] args) {
        SpringApplication.run(FlashcardsApplication.class, args);
        try {
            new ProcessBuilder("open", "http://localhost:8080").start();
        } catch (IOException e) {
            log.error("Failed to run browser", e);
        }
    }
}

File: src/main/kotlin/ru/uvarov/flashcards/FlashcardsApplication.kt
package ru.uvarov.flashcards

import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import java.io.IOException

@SpringBootApplication
class FlashcardsApplication

private val log = LoggerFactory.getLogger(FlashcardsApplication::class.java)

fun main(args: Array<String>) {
    runApplication<FlashcardsApplication>(*args)
    try {
        ProcessBuilder("open", "http://localhost:8080").start()
    } catch (e: IOException) {
        log.error("Failed to run browser", e)
    }
}
